python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -241,6 +232,26 @@ def __cuda_array_interface__(self) -> Mapping:
             "version": 0,
         }
 
+    @property
+    def exposed(self) -> bool:
+        """The current exposure status of the buffer
+
+        This is used by ExposureTrackedBuffer to determine when a deep copy
+        is required and by SpillableBuffer to mark the buffer unspillable.
+        """
+        return self._exposed
+
+    def mark_exposed(self) -> None:
```

Comment:
Any reason not to use `exposed.setter` here instead?

Reply:
To prevent `.exposed = False`, the expose mark is permanent.

Reply:
OK - that makes sense